File: accenture_python/maxsubarray.py
n, k = map(int, input().split())
arr = list(map(int, input().split()))

windowSum = sum(arr[:k])
maxSum = windowSum

for i in range(k, n):
    windowSum += arr[i]
    windowSum -= arr[i - k]

    maxSum = max(maxSum, windowSum)

print(maxSum)

File: slidingwindow/maxsubarray.cpp
#include<bits/stdc++.h>
class Solution {
public:
    int maxSumSubarray(vector<int>& arr, int k) {
        int n = arr.size();
        int windowSum = 0;
        for(int i =0;i<k;i++){
            windowSum+=arr[i];
        }
        maxSum = windowSum;

        for(int i = k;i<n;i++){
            windowSum+=arr[i];
            windowSum-=arr[i-k];

            maxSum = max(maxSum,windowSum);
        }

        return maxSum;  }
    };